### PR TITLE
Include tethys component

### LINKED ISSUE #15: Convert downscaling component to use tethys
The existing water downscaling component is a primitive version of [tethys](https://github.com/JGCRI/tethys).  Replace the old components with one that runs tethys.

### PR DESCRIPTION
**Pull request by calebbraun**
Add a water withdrawal downscaling component that calls tethys.

Resolve #15

### CODE REVIEW

**Comment by rplzzz** [comment]
It looks like this one is failing its tests because it can't locate the `gcam_reader` package.

File: cassandra/components.py [hunk 1]
```diff
@@ -522,6 +525,38 @@ def run_component(self):
...
+    For more information: https://github.com/JGCRI/tethys
+
+    params:
+       config_file - path to Tethys config file
+    """
+
+    def __init__(self, cap_tbl):
+        super(TethysComponent, self).__init__(cap_tbl)
+        self.addcapability("tethys_gis")
+        self.addcapability("tethys_gridded")
```
Comment (rplzzz) [inline cassandra/components.py]:
```suggestion
        self.addcapability('water_downscaling')
```

File: cassandra/components.py [hunk 2]
```diff
@@ -522,6 +525,38 @@ def run_component(self):
...
+        """Run Tethys."""
+        from tethys.model import Tethys
+
+        config_file = self.params["config_file"]
+
+        # run the Tethys model
+        tethys_results = Tethys(config=config_file)
+
+        self.addresults("tethys_gis", tethys_results.gis_data)
+        self.addresults("tethys_gridded", tethys_results.gridded_data)
```
Comment (rplzzz) [inline cassandra/components.py]:
```suggestion
        self.addresults('water_downscaling', tethys_results.gridded_data)
```

**Comment by rplzzz** [comment]
Gah! Failing because Xanthos doesn't support Python 3.6.  Is that being updated on the Xanthos side?

**Comment by rplzzz** [comment]
@calebbraun & @crvernon It also looks as though pip is still reporting `--process-dependency-links` as deprecated.  I wonder if we should just publish them in PyPi.  We don't normally publish to CRAN because its requirements can be a little onerous (particularly the max package size), but it looks like PyPi is a lot looser in that respect.  What do you guys think?

**Comment by calebbraun** [comment]
I like the idea of publishing our models to PyPi. I can see two problems that we would need to think about though. First, they don't support organizational accounts, so we would have to be very careful about keeping package ownership distributed among multiple people. Second, there seems to be a semi-flexible 60MB upload limit, meaning we probably couldn't ship example data.

**Pull request merged**